torchgeo/models/cnnlstm.py

```diff
@@ -0,0 +1,378 @@
+            "task_type must be 'classification' or 'regression'"
+        )
+        self.task_type = task_type
+        self.pooling = pooling
+        self.dropout = nn.Dropout(dropout)
+
+        if self.pooling == 'attention':
+            self.attention = nn.Linear(input_dim, 1)
+
+        if task_type == 'classification':
+            assert num_classes is not None, (
+                'num_classes must be specified for classification'
+            )
+            self.head = nn.Linear(input_dim, num_classes)
+        else:  # regression
+            self.head = nn.Linear(input_dim, output_dim)
+
+    def forward(self, x: torch.Tensor) -> dict[str, torch.Tensor | None]:
+        """Forward pass through task head.
+
+        Args:
+            x: Input tensor, either (B, input_dim) for single vector or
+               (B, T, input_dim) for sequences.
+
+        Returns:
+            Dictionary containing:
+                For classification:
+                    - 'outputs': Raw logits tensor (B, num_classes).
+                    - 'probs': Softmax probabilities tensor (B, num_classes).
+                For regression:
+                    - 'outputs': Regression predictions (B, output_dim).
+                    - 'probs': None.
+        """
+        # Apply pooling if input is sequence (B, T, input_dim)
+        if x.dim() == 3:
+            if self.pooling == 'last':
+                x = x[:, -1, :]
+            elif self.pooling == 'mean':
+                x = x.mean(dim=1)
+            elif self.pooling == 'max':
+                x = x.max(dim=1)[0]
+            elif self.pooling == 'attention':
+                attn_weights = torch.softmax(self.attention(x), dim=1)
+                x = (x * attn_weights).sum(dim=1)
+
+        x = self.dropout(x)
+        outputs = self.head(x)
+
+        if self.task_type == 'classification':
+            probs = torch.softmax(outputs, dim=-1)
+            return {'outputs': outputs, 'probs': probs}
+        else:  # regression
+            return {'outputs': outputs, 'probs': None}
+
+
+class CNNLSTM(nn.Module):
+    """Combined CNN + LSTM model with a unified task head.
+
+    Args:
+        cnn_backbone (CNN): CNN feature extractor module.
+        rnn_encoder (LSTM): LSTM sequence encoder module.
+        task_head (Task): Task head for classification or regression.
+
+    Raises:
+        ValueError: If the output dimension of CNN does not match input dimension of LSTM,
+                    or if the output dimension of LSTM does not match input dimension of task head.
+
+    Forward:
+        x (torch.Tensor): Input tensor, either
+            - (B, T, C, H, W) sequence of images, or
+            - (B, C, H, W) single image.
+        lengths (torch.Tensor, optional): Lengths of sequences for LSTM packing.
+
+    Returns:
+        Dict[str, Optional[torch.Tensor]]: Output dictionary from task head.
+    """
+
+    def __init__(self, cnn_backbone: CNN, rnn_encoder: LSTM, task_head: Task) -> None:
```

Comment:
We want this to be usable via YAML configuration file. So strings need to be supported. We _could_ support nn.Modules too, but if strings work, then nn.Modules probably aren't needed.